### ISSUE TITLE
Only 1st tag is applied

### ISSUE THREAD

**Issue by shompoe** [etesync/etesync-web#171]
Create a task in Tasks.org and apply more than one tag. Synchronise up and look in the web client. The task is shown only under the first tag but not the others.

**Comment by tasn** [comment]
Could you please paste here (feel free to censor here) the raw content of the task in question? You can do it opening the task in the web client, click on it -> change history -> the most recent change. Paste it here so we can see if the issue is in Tasks.org or the etesync web client.

Thanks!

**Comment by shompoe** [comment]
Just to clarify, since reseting the phone, solving sync issues, and re-creating the tasks list, this behaviour is still occuring.

Can you not reproduce it?
--- 
Mark Desrousseaux



18 Oct 2020, 08:01 by notifications@github.com:

**Comment by tasn** [comment]
I'd expect that. My request still stands. Could you please paste here the raw content of the task in question as requested above?

**Comment by shompoe** [comment]
Tom,

I've solved it.

I've had so many issues with tasks going back so long, which I now know all originated when something got messed up in my phone.

This morning I noticed deleted tasks were not appearing in the journal again.

So I deleted task data from tasks and made new journals again. Then I edited my tasks.org backup file before importing it again, but this time I removed etag and set lastsync to 0 for every task - so they would look like they had been freshly created in tasks app.

Now when I delete tasks they stay deleted and when I log into web client everything is the same as in tasks app. Nothing is hidden, tags are correctly populated, etc.

Sorry to have bothered you. And it's reassuring to know that the apps are fine - it was all caused by device fault from the very begining.
--- 
Mark Desrousseaux



19 Oct 2020, 06:50 by notifications@github.com:

**Comment by shompoe** [comment]
Sorry, I thought I had sent it.
#########################################################

Entry UID:
70a37ddb72e5dc5f03643ac23d1e3dfe6193884f93919bb08c86adf183edb046
Content:
BEGIN:VCALENDAR
VERSION:2.0
PRODID:-//EteSync//com.etesync.syncadapter/1.16.2 ical4j/2.2.6
BEGIN:VTODO
DTSTAMP:20201020T103626Z
UID:2030497105674122228
CREATED:20201020T103619Z
LAST-MODIFIED:20201020T103619Z
SUMMARY:Check hours
PRIORITY:9
STATUS:NEEDS-ACTION
RRULE:FREQ=WEEKLY;INTERVAL=1
CATEGORIES:Finance,Hide from week view until due date
DUE;TZID=Europe/London:20201022T190001
END:VTODO
BEGIN:VTIMEZONE
TZID:Europe/London
TZURL:http://tzurl.org/zoneinfo/Europe/London
X-LIC-LOCATION:Europe/London
BEGIN:DAYLIGHT
TZOFFSETFROM:+0000
TZOFFSETTO:+0100
TZNAME:BST
DTSTART:19810329T010000
RRULE:FREQ=YEARLY;BYMONTH=3;BYDAY=-1SU
END:DAYLIGHT
BEGIN:STANDARD
TZOFFSETFROM:+0100
TZOFFSETTO:+0000
TZNAME:GMT
DTSTART:19961027T020000
RRULE:FREQ=YEARLY;BYMONTH=10;BYDAY=-1SU
END:STANDARD
BEGIN:STANDARD
TZOFFSETFROM:-000115
TZOFFSETTO:+0000
TZNAME:GMT
DTSTART:18471201T000000
RDATE:18471201T000000
END:STANDARD
BEGIN:DAYLIGHT
TZOFFSETFROM:+0000
TZOFFSETTO:+0100
TZNAME:BST
DTSTART:19160521T020000
RDATE:19160521T020000
RDATE:19170408T020000
RDATE:19180324T020000
RDATE:19190330T020000
RDATE:19200328T020000
RDATE:19210403T020000
RDATE:19220326T020000
RDATE:19230422T020000
RDATE:19240413T020000
RDATE:19250419T020000
RDATE:19260418T020000
RDATE:19270410T020000
RDATE:19280422T020000
RDATE:19290421T020000
RDATE:19300413T020000
RDATE:19310419T020000
RDATE:19320417T020000
RDATE:19330409T020000
RDATE:19340422T020000
RDATE:19350414T020000
RDATE:19360419T020000
RDATE:19370418T020000
RDATE:19380410T020000
RDATE:19390416T020000
RDATE:19400225T020000
RDATE:19460414T020000
RDATE:19470316T020000
RDATE:19480314T020000
RDATE:19490403T020000
RDATE:19500416T020000
RDATE:19510415T020000
RDATE:19520420T020000
RDATE:19530419T020000
RDATE:19540411T020000
RDATE:19550417T020000
RDATE:19560422T020000
RDATE:19570414T020000
RDATE:19580420T020000
RDATE:19590419T020000
RDATE:19600410T020000
RDATE:19610326T020000
RDATE:19620325T020000
RDATE:19630331T020000
RDATE:19640322T020000
RDATE:19650321T020000
RDATE:19660320T020000
RDATE:19670319T020000
RDATE:19680218T020000
RDATE:19720319T020000
RDATE:19730318T020000
RDATE:19740317T020000
RDATE:19750316T020000
RDATE:19760321T020000
RDATE:19770320T020000
RDATE:19780319T020000
RDATE:19790318T020000
RDATE:19800316T020000
END:DAYLIGHT
BEGIN:STANDARD
TZOFFSETFROM:+0100
TZOFFSETTO:+0000
TZNAME:GMT
DTSTART:19161001T030000
RDATE:19161001T030000
RDATE:19170917T030000
RDATE:19180930T030000
RDATE:19190929T030000
RDATE:19201025T030000
RDATE:19211003T030000
RDATE:19221008T030000
RDATE:19230916T030000
RDATE:19240921T030000
RDATE:19251004T030000
RDATE:19261003T030000
RDATE:19271002T030000
RDATE:19281007T030000
RDATE:19291006T030000
RDATE:19301005T030000
RDATE:19311004T030000
RDATE:19321002T030000
RDATE:19331008T030000
RDATE:19341007T030000
RDATE:19351006T030000
RDATE:19361004T030000
RDATE:19371003T030000
RDATE:19381002T030000
RDATE:19391119T030000
RDATE:19451007T030000
RDATE:19461006T030000
RDATE:19471102T030000
RDATE:19481031T030000
RDATE:19491030T030000
RDATE:19501022T030000
RDATE:19511021T030000
RDATE:19521026T030000
RDATE:19531004T030000
RDATE:19541003T030000
RDATE:19551002T030000
RDATE:19561007T030000
RDATE:19571006T030000
RDATE:19581005T030000
RDATE:19591004T030000
RDATE:19601002T030000
RDATE:19611029T030000
RDATE:19621028T030000
RDATE:19631027T030000
RDATE:19641025T030000
RDATE:19651024T030000
RDATE:19661023T030000
RDATE:19671029T030000
RDATE:19711031T030000
RDATE:19721029T030000
RDATE:19731028T030000
RDATE:19741027T030000
RDATE:19751026T030000
RDATE:19761024T030000
RDATE:19771023T030000
RDATE:19781029T030000
RDATE:19791028T030000
RDATE:19801026T030000
RDATE:19811025T020000
RDATE:19821024T020000
RDATE:19831023T020000
RDATE:19841028T020000
RDATE:19851027T020000
RDATE:19861026T020000
RDATE:19871025T020000
RDATE:19881023T020000
RDATE:19891029T020000
RDATE:19901028T020000
RDATE:19911027T020000
RDATE:19921025T020000
RDATE:19931024T020000
RDATE:19941023T020000
RDATE:19951022T020000
END:STANDARD
BEGIN:DAYLIGHT
TZOFFSETFROM:+0100
TZOFFSETTO:+0200
TZNAME:BDST
DTSTART:19410504T010000
RDATE:19410504T010000
RDATE:19420405T010000
RDATE:19430404T010000
RDATE:19440402T010000
RDATE:19450402T010000
RDATE:19470413T010000
END:DAYLIGHT
BEGIN:DAYLIGHT
TZOFFSETFROM:+0200
TZOFFSETTO:+0100
TZNAME:BST
DTSTART:19410810T030000
RDATE:19410810T030000
RDATE:19420809T030000
RDATE:19430815T030000
RDATE:19440917T030000
RDATE:19450715T030000
RDATE:19470810T030000
END:DAYLIGHT
BEGIN:STANDARD
TZOFFSETFROM:+0100
TZOFFSETTO:+0100
TZNAME:BST
DTSTART:19681026T230000
RDATE:19681026T230000
END:STANDARD
BEGIN:STANDARD
TZOFFSETFROM:+0000
TZOFFSETTO:+0000
TZNAME:GMT
DTSTART:19960101T000000
RDATE:19960101T000000
END:STANDARD
END:VTIMEZONE
END:VCALENDAR

**Comment by shompoe** [comment]
In the raw data there are two tags but if I open the task in web client it only lists the first one.

**Comment by shompoe** [comment]
OK, so if I create a task with multiple tags from within the web client, then it is displayed correctly under those tags in the web client, but when it syncs back to tasks.org, it appears under a new tag, which is concatenated from the individual tag names separated with commas.

Example:    
web client Tag1: apples    Tag2: oranges
tasks.org   Tag1: apples, oranges

**Comment by tasn** [comment]
That's progress and may explain what is going on. When you say tasks.org: are you using the built-in tasks.org etesync support (that is, enter your credentials there), or the one that integrates with the etesync app?

**Comment by shompoe** [comment]
Integrating with etesync app now.
--- 
Mark Desrousseaux



20 Oct 2020, 13:03 by notifications@github.com:

**Comment by tasn** [comment]
Could you please share the raw entry of a multi-tag task created from Tasks.org? Thanks!

**Comment by shompoe** [comment]
Yep.     That's what I shared above.

**Comment by tasn** [comment]
Can you please share the other one then? The one that was working in the web client but not in tasks.org?

**Comment by shompoe** [comment]
Entry UID:
9339b60aa02de4777f14628a42e500e922621c51f0d6ff56e93ac497928afaaf
Content:
BEGIN:VCALENDAR
VERSION:2.0
PRODID:+//IDN bitfire.at//ical4android
BEGIN:VTODO
DTSTAMP:20201020T113337Z
UID:eafa21d4-19b0-4a23-900f-2e97e9080b4c
LAST-MODIFIED:20201020T113123Z
SUMMARY:Test
STATUS:COMPLETED
CATEGORIES:Finance\,House and Garden
END:VTODO
END:VCALENDAR


Comma is escaped out?

**Comment by tasn** [comment]
I don't see an escape (maybe it's because you are not pasting it as part of a code tag? could you maybe edit your posts), here is what I'm seeing:
```
 CATEGORIES:Finance,Hide from week view until due date
vs
 CATEGORIES:Finance,House and Garden
```
Looks the same (with regarding to the commas), no?

**Comment by shompoe** [comment]
Entry UID:
aa7297141be8a5638569ec5af1f62af1cee42317867fcd4dcceeb402a58666c4
Content:
BEGIN:VCALENDAR
PRODID:-//iCal.js EteSync iOS
VERSION:2.0
BEGIN:VTODO
UID:4e6c3d6a-8463-4a32-8851-e466bb2bcf9f
SUMMARY:Test2
STATUS:NEEDS-ACTION
PRIORITY:0
CATEGORIES:Car\,Shopping
LOCATION:
DESCRIPTION:
LAST-MODIFIED:20201020T141754
END:VTODO
END:VCALENDAR

**Comment by shompoe** [comment]
Strange,      Github is removing the backslash before the comma when I commit the comment

**Comment by tasn** [comment]
I edited your posts to enclose your pastes in code blocks, which is then not escaped by Github, and I can now see the issue. Seems to be a bug in the web app indeed.

**Comment by shompoe** [comment]
At last I've been helpful as opposed to wasting your time!  Lol.
--- 
Mark Desrousseaux



21 Oct 2020, 09:45 by notifications@github.com:

**Comment by tasn** [comment]
I just deployed a new version to pim.etesync.com (the EteSync 2.0 one!). Could you please check to see if that fixes it for you? You may need to force refresh the page to make sure you get the latest version, and you may need to create a new task/edit the tags to make sure old ones are fixed.

**Comment by tasn** [comment]
Edit: actually, deployed it to the legacy one too.

**Comment by shompoe** [comment]
Uh oh.....      As soon as i click over from default calendar view to tasks I get this........

```
Something went wrong!
Cannot read property 'getValues' of null
TypeError: Cannot read property 'getValues' of null
    at t.get (https://client.etesync.com/static/js/main.349de1e8.chunk.js:1:150258)
    at https://client.etesync.com/static/js/main.349de1e8.chunk.js:1:229997
    at Array.filter (<anonymous>)
    at Qa (https://client.etesync.com/static/js/main.349de1e8.chunk.js:1:229968)
    at Jo (https://client.etesync.com/static/js/2.012435f0.chunk.js:2:1378751)
    at gs (https://client.etesync.com/static/js/2.012435f0.chunk.js:2:1425157)
    at uc (https://client.etesync.com/static/js/2.012435f0.chunk.js:2:1417690)
    at sc (https://client.etesync.com/static/js/2.012435f0.chunk.js:2:1417615)
    at $s (https://client.etesync.com/static/js/2.012435f0.chunk.js:2:1414624)
    at https://client.etesync.com/static/js/2.012435f0.chunk.js:2:1366009
```

**Comment by shompoe** [comment]
Same on Etesync 2.0......

```
Something went wrong!
Cannot read property 'getValues' of null
TypeError: Cannot read property 'getValues' of null
    at t.get (https://pim.etesync.com/static/js/main.f1e22785.chunk.js:1:146343)
    at https://pim.etesync.com/static/js/main.f1e22785.chunk.js:1:210406
    at Array.forEach (<anonymous>)
    at https://pim.etesync.com/static/js/main.f1e22785.chunk.js:1:210376
    at Jo (https://pim.etesync.com/static/js/2.e77744e8.chunk.js:2:1304581)
    at Ra (https://pim.etesync.com/static/js/2.e77744e8.chunk.js:2:1313485)
    at Ta (https://pim.etesync.com/static/js/2.e77744e8.chunk.js:2:1313304)
    at Oa (https://pim.etesync.com/static/js/2.e77744e8.chunk.js:2:1312987)
    at gs (https://pim.etesync.com/static/js/2.e77744e8.chunk.js:2:1354535)
    at uc (https://pim.etesync.com/static/js/2.e77744e8.chunk.js:2:1343520)
```

**Comment by tasn** [comment]
My bad, please try again.

**Comment by shompoe** [comment]
All of my tasks created in tasks.org are now displaying correctly in web client.
However, when I now create a new task in the web client, it's tags are not getting saved. I looked at the raw task data to verify this.

**Comment by tasn** [comment]
Sorry for the sloppiness, I'm doing a few things in parallel. It should be good now. I hope. :)

**Comment by shompoe** [comment]
Hmmmmmm, nope, still not right I'm afraid.

<img width="438" alt="error" src="https://user-images.githubusercontent.com/54352351/96742160-7ef8a000-13ba-11eb-9807-4ac61c2cb4a4.png">

**Comment by tasn** [comment]
OK, I'll stop wasting your time, hehe. I'm on it. :P

**Comment by tasn** [comment]
I actually tested it now and it works. Please report back. :)

**Comment by shompoe** [comment]
Touché!

By the way, is it intended that when you restore state from the journal in web client it restores everything from the selected entry to the latest entry? Because is android client you can restore state of individual entries on their own, which is the way I would have expected it to work.

**Comment by tasn** [comment]
OK, so works? Can I push the changes?

The "restore state" doesn't restore the state of a specific item, but rather tries to rollback all of the changes in the list up until that item. It's a bit different than to Android, but it was removed in EteSync 2.0 (at least should have been).

**Comment by shompoe** [comment]
Not sure,   when I click "Save" the save button briefly flashes but then nothing happens at all, so still can't create any tasks.

**Comment by tasn** [comment]
So weird, as it works here. :|
Are you testing with etesync v1, v2 or both?

**Comment by shompoe** [comment]
Version 1.0 (in brave browser)

**Comment by shompoe** [comment]
V2 is also the same.

**Comment by tasn** [comment]
Ahhh, it's my bad actually, I tested the wrong thing. Fixing and testing again.

**Comment by tasn** [comment]
OK, I now tested the right thing. I previously tested editing a task with existing tags. I now also tested creating tags for event without any. :)

**Comment by shompoe** [comment]
Looking good.

Thanks for fixing. If I notice anything else I'll let you know.

**Comment by tasn** [comment]
Great! And sorry for the ordeal. As I said, I was juggling a lot of things today in parallel and didn't have time to properly test things. Ended up costing more time in the end. As such things always do... :)
